x-pack/plugins/apm/server/lib/observability_dashboard/get_transaction_coordinates.ts

```diff
@@ -11,20 +11,27 @@
 import { rangeFilter } from '../../../common/utils/range_filter';
 import { Coordinates } from '../../../../observability/public';
 import { Setup, SetupTimeRange } from '../helpers/setup_request';
-import { ProcessorEvent } from '../../../common/processor_event';
+import {
+  getProcessorEventForAggregatedTransactions,
+  getTransactionDurationFieldForAggregatedTransactions,
+} from '../helpers/aggregated_transactions/get_use_aggregated_transaction';
 
 export async function getTransactionCoordinates({
   setup,
   bucketSize,
+  useAggregatedTransactions,
 }: {
   setup: Setup & SetupTimeRange;
   bucketSize: string;
+  useAggregatedTransactions: boolean;
 }): Promise<Coordinates[]> {
   const { apmEventClient, start, end } = setup;
 
   const { aggregations } = await apmEventClient.search({
     apm: {
-      events: [ProcessorEvent.transaction],
+      events: [
+        getProcessorEventForAggregatedTransactions(useAggregatedTransactions),
```

Comment:
Would the shorter version be ok?
```suggestion
        getProcessorEvent(useAggregatedTransactions),
```

or at least this would be more correct imo (since it may or may not be aggregated - but we know for sure it's a transaction)
```suggestion
        getTransactionProcessorEvent(useAggregatedTransactions),
```